crates/noirc_evaluator/src/ssa_refactor/ir/basic_block.rs

```diff
@@ -52,7 +52,29 @@ impl BasicBlock {
         self.instructions.push(instruction);
     }
 
+    pub(crate) fn instructions(&self) -> &[InstructionId] {
+        &self.instructions
+    }
+
     pub(crate) fn set_terminator(&mut self, terminator: TerminatorInstruction) {
         self.terminator = Some(terminator);
     }
+
+    pub(crate) fn terminator(&self) -> Option<&TerminatorInstruction> {
+        self.terminator.as_ref()
+    }
+
+    /// Iterate over all the successors of the currently block, as determined by
+    /// the blocks jumped to in the terminator instruction. If there is no terminator
+    /// instruction yet, this will iterate 0 times.
+    pub(crate) fn successors(&self) -> impl ExactSizeIterator<Item = BasicBlockId> {
```

Comment:
This function is redundant with the PR https://github.com/noir-lang/noir/pull/1200. Once the former is merged, this will need to be changed.

Reply:
I've made an issue for this here: https://github.com/noir-lang/noir/issues/1212 and assigned it to @joss-aztec